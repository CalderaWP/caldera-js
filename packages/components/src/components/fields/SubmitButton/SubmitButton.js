import {FormTokenField} from '@wordpress/components';
import { BaseControl } from '@wordpress/components';
import React from "react";
import {fieldClassNames, parseAttributes} from "../util";
export const SubmitButton = (props) => {
	const {
		label,
		description,
		fieldId,
		placeholder,
		required,
		value,
		onChange,
		onBlur,
		options,
		attributes
	} = props;
	const _attributes = {
		disabled: 'object' === typeof  attributes
			&& attributes.hasOwnProperty('disabled')
			&& true == attributes.disabled,
	}
	const {disabled} = _attributes;
	return (
		<BaseControl
			id={fieldId}
			help={description}
		>
			<input
				type={'submit'}
				id={fieldId}
				value={label}
				disabled={disabled}
			/>
		</BaseControl>
	);
};

