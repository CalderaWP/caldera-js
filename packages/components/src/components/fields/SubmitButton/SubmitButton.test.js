import * as React from 'react';
import renderer from 'react-test-renderer';

import { SubmitButton } from './SubmitButton';

describe('Select Field component', () => {
	let onChange;
	beforeEach(() => {
		onChange = jest.fn();
	});
	it('matches snapshot', () => {
		const component = renderer.create(
			<SubmitButton
				label={'Send Message'}
				description={'Click to sendu'}
				fieldId={'button'}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});


});
