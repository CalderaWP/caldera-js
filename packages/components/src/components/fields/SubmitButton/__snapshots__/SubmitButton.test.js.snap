// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Field component matches snapshot 1`] = `
<div
  className="components-base-control"
>
  <div
    className="components-base-control__field"
  >
    <input
      disabled={false}
      id="button"
      type="submit"
      value="Send Message"
    />
  </div>
  <p
    className="components-base-control__help"
    id="button__help"
  >
    Click to sendu
  </p>
</div>
`;
